### PR TITLE
fix #1479 allow to override SubresourceOperations through metadata

### LINKED ISSUE #1479: How to change Subresource path
Is it possible to change the path of a subresource?

### PR DESCRIPTION
**Pull request by soyuka**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #1479 
| License       | MIT
| Doc PR        | TODO

Also backported #1472 to subresources

### CODE REVIEW

File: src/Bridge/Symfony/Routing/ApiLoader.php [hunk 1]
```diff
@@ -87,6 +87,9 @@ public function load($data, $type = null): RouteCollection
 
             if (null !== $collectionOperations = $resourceMetadata->getCollectionOperations()) {
                 foreach ($collectionOperations as $operationName => $operation) {
+                    if ('subresource' === substr($operationName, -11)) {
```
Comment (soyuka) [inline src/Bridge/Symfony/Routing/ApiLoader.php]: Either this or we add a `SubresourceOperations` in the resource metadata declarations. Lmk.

Comment (iCodr8) [reply src/Bridge/Symfony/Routing/ApiLoader.php]: I think SubresourceOperations would make more sense.

Comment (soyuka) [reply src/Bridge/Symfony/Routing/ApiLoader.php]: done

File: src/Metadata/Resource/ResourceMetadata.php [hunk 2]
```diff
@@ -163,6 +165,31 @@ public function withCollectionOperations(array $collectionOperations): self
...
+     */
+    public function getSubresourceOperations()
+    {
+        return $this->subresourceOperations;
+    }
+
+    /**
+     * Returns a new instance with the given subresource operations.
+     *
+     * @param array $subresourceOperations
```
Comment (dunglas) [inline src/Metadata/Resource/ResourceMetadata.php]: You can remove those annotations (it's already documented by typehints).

File: src/Metadata/Resource/ResourceMetadata.php [hunk 3]
```diff
@@ -193,6 +220,21 @@ public function getItemOperationAttribute(string $operationName = null, string $
     }
 
     /**
+     * Gets a subresource operation attribute, optionally fallback to a resource attribute.
+     *
+     * @param string|null $operationName
```
Comment (dunglas) [inline src/Metadata/Resource/ResourceMetadata.php]: Same here

File: src/Bridge/Symfony/Routing/ApiLoader.php (hunk 1 above)
Comment (iCodr8) [reply src/Bridge/Symfony/Routing/ApiLoader.php]: Nice, thank you!

File: src/Metadata/Resource/ResourceMetadata.php (hunk 2 above)
Comment (soyuka) [reply src/Metadata/Resource/ResourceMetadata.php]: copy pasted, why don't we remove the other ones then?

File: src/Bridge/Symfony/Routing/ApiLoader.php [hunk 4]
```diff
@@ -115,12 +115,13 @@ public function load($data, $type = null): RouteCollection
                             'collection' => $operation['collection'],
                             'operationId' => $operationId,
                         ],
-                    ],
-                    $operation['requirements'] ?? [],
-                    [],
-                    '',
-                    [],
-                    ['GET']
+                    ] + ($operation['defaults'] ?? []),
```
Comment (meyerbaptiste) [inline src/Bridge/Symfony/Routing/ApiLoader.php]: All parenthesis are useless here.

**Comment by meyerbaptiste** [comment]
Thanks @soyuka

**Pull request merged**